### PR TITLE
[ISSUE-3830][Feature] Introduce upload management e2e test

### LINKED ISSUE #3830: [Feature] Add resource upload e2e test cases
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Add Resource Upload e2e test cases:
-  createResource
- deleteResource
- editResource

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengkezhou1**
## What changes were proposed in this pull request

Issue Number: close #3830 

add basic resource management test cases.
## Verifying this change

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (no)

### CODE REVIEW

**Comment by zhengkezhou1** [comment]
@SbloodyS Many problems here and difficult to solve for me. Can you take a look thanks?

**Comment by SbloodyS** [comment]
> @SbloodyS Many problems here and difficult to solve for me. Can you take a look thanks?

Sure. What's the problem?

**Comment by zhengkezhou1** [comment]
> > @SbloodyS Many problems here and difficult to solve for me. Can you take a look thanks?
> 
> Sure. What's the problem?

Thanks for reply, which was previously marked as `TODO` before and has now been resolved.

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/resource/ResourceManagementTest.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                () -> assertThat(resourceManagementPage.resourceList())
+                    .as("Resource list should contain newly-created resource")
+                    .extracting(WebElement::getText)
+                    .anyMatch(it -> it.contains(resourceName))
+                    .anyMatch(it -> it.contains(description))
+                    /*
+                     * todo: make Resource Type filed value same, hardcode here to pass test currently. In add resource
+                     * page: Add Resource -> Resource Type: Jar Library In upload page: Resource List -> Resource Type:
+                     * Jar library
+                     */
```
Comment (zhengkezhou1) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/resource/ResourceManagementTest.java]: I will try to improve the frontend soon .

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/resource/ResourceManagementTest.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                () -> assertThat(resourceManagementPage.resourceList())
+                    .as("Resource list should contain edit resource")
+                    .extracting(WebElement::getText)
+                    .anyMatch(it -> it.contains(resourceName))
+                    .anyMatch(it -> it.contains(editDescription))
+                    /*
+                     * todo: make Resource Type filed value same, hardcode here to pass test currently. In add resource
+                     * page: Add Resource -> Resource Type: Jar Library In upload page: Resource List -> Resource Type:
+                     * Jar library
+                     */
```
Comment (zhengkezhou1) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/resource/ResourceManagementTest.java]: I will try to improve the frontend soon .

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -105,10 +105,12 @@ jobs:
     strategy:
       matrix:
         case:
-          - name: MemberManagementTest
-            class: org.apache.streampark.e2e.cases.MemberManagementTest
+          - name: ResourceManagementTest.java
+            class: org.apache.streampark.e2e.cases.resource.ResourceManagementTest
```
Comment (SbloodyS) [inline .github/workflows/e2e.yml]: We don't need to add subdirectory here.

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -105,10 +105,12 @@ jobs:
     strategy:
       matrix:
         case:
-          - name: MemberManagementTest
-            class: org.apache.streampark.e2e.cases.MemberManagementTest
+          - name: ResourceManagementTest.java
```
Comment (SbloodyS) [inline .github/workflows/e2e.yml]:
```suggestion
          - name: ResourceManagementTest
```

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/resource/ResourceManagementTest.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@StreamPark(composeFiles = "docker/basic/docker-compose.yaml")
+public class ResourceManagementTest {
```
Comment (SbloodyS) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/cases/resource/ResourceManagementTest.java]:
```suggestion
public class UploadManagementTest {
```

We should be consistent with the page name.

File: streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/pages/resource/UploadManagementPage.java [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    private WebElement errorMessageConfirmButton;
+
+    @FindBy(xpath = "//button[contains(@class, 'ant-btn')]/span[contains(., 'OK')]")
+    private WebElement deleteConfirmButton;
+
+    public UploadManagementPage(RemoteWebDriver driver) {
+        super(driver);
+    }
+
+    public UploadManagementPage createResource(String engineType, String resourceType, String resourceName,
```
Comment (SbloodyS) [inline streampark-e2e/streampark-e2e-case/src/test/java/org/apache/streampark/e2e/pages/resource/UploadManagementPage.java]:
```suggestion
    public UploadManagementPage createUpload(String engineType, String resourceType, String resourceName,
```

**Review by SbloodyS — COMMENTED** [review]
We should change all this to upload.

**Review by SbloodyS — APPROVED** [review]
LGTM if CI passed. PTAL @wolfboys

**Review by wolfboys — APPROVED** [review]

**Pull request merged**